WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift

```diff
@@ -41,10 +41,25 @@ public final class SingleProductScreen: ScreenObject {
         return self
     }
 
-    public func verifyNewProductScreenLoaded(productName: String) {
+    public func verifyPublishedProductScreenLoaded(productType: String, productName: String) {
+        let priceLabel = NSPredicate(format: "label == 'Price'")
+        let addVariationLabel = NSPredicate(format: "label == 'Add variations'")
+
+        // common fields on a published product screen
         XCTAssertTrue(app.buttons["save-product-button"].exists)
         XCTAssertTrue(app.staticTexts["TIP"].exists)
         XCTAssertTrue(app.textViews[productName].exists)
+
+        // different product types display different fields on the published product screen
+        // this is to validate that the correct screens are displayed
+        switch productType {
+        case "physical", "virtual":
+            XCTAssertTrue(app.staticTexts.containing(priceLabel).firstMatch.exists)
+        case "variable":
+            XCTAssertTrue(app.staticTexts.containing(addVariationLabel).firstMatch.exists)
```

Comment:
Maybe overthinking, but additionally checking for the absence of elements might improve robustness a bit (just an idea):

```suggestion
        case "physical", "virtual":
            XCTAssertTrue(app.staticTexts["Price"].exists)
            XCTAssertFalse(app.staticTexts["Add variations"].exists)
        case "variable":
            XCTAssertTrue(app.staticTexts["Add variations"].exists)
            XCTAssertFalse(app.staticTexts["Price"].exists)
```

Reply:
thanks for thinking about this too! but i think i'll keep only the true assert for now. adding the false assert will add extra run time to the test and i question the value that it adds with the added time since we're already asserting that the unique cells are visible on the screen